clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -5193,12 +5193,12 @@ bool Sema::CheckCallingConvAttr(const ParsedAttr &Attrs, CallingConv &CC,
     CC = CC_X86RegCall;
     break;
   case ParsedAttr::AT_MSABI:
-    CC = Context.getTargetInfo().getTriple().isOSWindows() ? CC_C :
-                                                             CC_Win64;
+    CC = Context.getTargetInfo().getTriple().isOSWindowsOrUEFI() ? CC_C
```

Comment:
This really highlights how `getTargetInfo().getCallingConv()` is what we ought to have.  For now it can be defined just this way, but in future it might be controlled by switches within a given target like UEFI.  But right now, it makes code like this and `checkVAStartABI` much clearer.

Reply:
I'm not sure your comment makes sense in the context of the `__attribute__((msabi))` implementation. I would interpret your suggestion instead as something like `getTargetInfo().getCcForMsAbiAttribute()`, and the target would return the appropriate convention.

I think in general this will only matter on Win / non-Win x64 platforms where we have divergence between the Win/Posix worlds, while the are aligned in the ARM world.